drivers/ethernet/eth_stm32_hal.c

```diff
@@ -294,6 +363,41 @@ static int eth_initialize(struct device *dev)
 	return 0;
 }
 
+#if defined(CONFIG_PTP_CLOCK_STM32)
+static void ETH_PTPStart(ETH_HandleTypeDef *heth, uint32_t UpdateMethod) 
+{
+        
+        /* Check the parameters */
+        assert_param(IS_ETH_PTP_UPDATE(UpdateMethod));
+
+        /* Mask the Time stamp trigger interrupt by setting bit 9 in the MACIMR register. */
+        ETH_MACITConfig(heth, ETH_MAC_IT_TST, DISABLE);
+        
+        ETH_PTPTimeStampCmd(heth, ENABLE);
+
+        /* Program the Subsecond increment register based on the PTP clock frequency. */
+        ETH_SetPTPSubSecondIncrement(heth, ETH_STM32_PTP_INCREMENT);
+
+        if (UpdateMethod == ETH_PTP_FineUpdate) {
+                /* If you are using the Fine correction method, program the Time stamp addend register
+                * and set Time stamp control register bit 5 (addend register update). */
+                ETH_SetPTPTimeStampAddend(heth, ETH_STM32_PTP_ADDEND);
+                ETH_EnablePTPTimeStampAddend(heth);
+                /* Poll the Time stamp control register until bit 5 is cleared. */
+                while(ETH_GetPTPFlagStatus(heth, ETH_PTP_FLAG_TSARU) == SET);
+        }
+
+        /* To select the Fine correction method (if required),
+        * program Time stamp control register  bit 1. */
```

Comment:
Place the */ to next line, so multi line comments should look like this

```
/* This is a comment
 * second line
 */
```

Please check all the multi line comments in this file.